### ISSUE TITLE
Not building on MacOS High Sierra

### ISSUE THREAD

**Issue by colford** [bridgecommand/bc#4]
After running:

xcodebuild -project Irrlicht.xcodeproj

I see the following error:

/Users/gpsadmin/Src/bc/libs/Irrlicht/irrlicht-svn/source/Irrlicht/CIrrDeviceOSX.mm:581:23: error: cannot initialize a parameter of type
      'id<NSApplicationDelegate> _Nullable' with an rvalue of type 'id<NSFileManagerDelegate>'
                        [NSApp setDelegate:(id<NSFileManagerDelegate>)[[[CIrrDelegateOSX alloc] initWithDevice:this] autorelease]];

**Comment by SebastianKuehl** [comment]
I have a similar error:

```
/Users/basti/Documents/bridgecommand2/libs/Irrlicht/irrlicht-1.8.4/source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm:499:23: error: 
      cannot initialize a parameter of type
      'id<NSApplicationDelegate> _Nullable' with an rvalue of type
      'id<NSFileManagerDelegate>'
  ...(id<NSFileManagerDelegate>)[[[AppDelegate alloc] initWithDevice:this] autorelease]];
```

I could fix it by changing NSFileManagerDelegate in ClrrDeviceMacOSX.mm to NSApplicationDelegate in the specific location. I tried to continue with the next step: make. I got:

```
Undefined symbols for architecture x86_64:
  "NumberToImage::getImage(unsigned int, irr::IrrlichtDevice*)", referenced from:
      RadarCalculation::render(irr::video::IImage*, irr::video::IImage*, float, float) in RadarCalculation-8cb35d.o
  "MyEventReceiver::MyEventReceiver(irr::IrrlichtDevice*, SimulationModel*, GUIMain*, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >*)", referenced from:
      _main in main-cd9877.o
  "vtable for MyEventReceiver", referenced from:
      _main in main-cd9877.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

But this is weird, because I checked all the mentioned namespaces and functions and everything seems fine.. I dont know what is wrong. The codeblocks project file doesnt work, using terminal doesnt work.. has anyone a working project for macos high sierra?

**Comment by bridgecommand** [comment]
This should be fixed by be7946661c225a3b8976f9319da3ce60b7589e2d
